metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -605,6 +605,16 @@ ControllerResult<CreateTopicsResponseData> createTopics(
         Map<ConfigResource, Map<String, Entry<OpType, String>>> configChanges =
             computeConfigChanges(topicErrors, request.topics());
 
+        int totalPartitions = 0;
+        for (CreatableTopic topic: request.topics()) {
+            if (topic.numPartitions() > 0) {
+                totalPartitions += topic.numPartitions();
```

Comment:
This isn't quite right since we allow -1 as a special value for `numPartitions`. It means that we are explicitly specifying the partitions we want to create. These partitions should be counted towards the total too, of course.

Reply:
I see! Left a comment on new code with description of approach.